app/services/supplementary-billing/fetch-invoices-to-be-reissued.service.js

```diff
@@ -0,0 +1,58 @@
+'use strict'
+
+/**
+ * Fetches billing invoices to be reissued
+ * @module FetchInvoicesToBeReissuedService
+ */
+
+const BillingInvoiceModel = require('../../models/water/billing-invoice.model.js')
+
+/**
+ * Takes a region and fetches billing invoices in that region marked for reissuing, along with their transactions
+ *
+ * @param {String} regionId The uuid of the region
+ *
+ * @returns {module:BillingInvoiceModel[]} Array of billing invoices to be reissued
+ */
+async function go (regionId) {
+  try {
+    const result = await BillingInvoiceModel.query()
+      .select(
+        'billingInvoices.externalId',
+        'invoiceAccountId',
+        'invoiceAccountNumber',
+        'originalBillingInvoiceId'
+      )
+      .where('isFlaggedForRebilling', true)
+      .joinRelated('billingBatch')
+      .where('billingBatch.regionId', regionId)
+      .withGraphFetched('billingInvoiceLicences.billingTransactions')
+      .modifyGraph('billingInvoiceLicences', (builder) => {
+        builder.select(
+          'licenceRef',
+          'licenceId'
+        )
+      })
+
+    return result
+  } catch (error) {
+    // If getting invoices errors then we log the error and return an empty array; the db hasn't yet been modified at
+    // this stage so we can simply move on to the next stage of processing the billing batch.
+
+    global.GlobalNotifier.omfg('Could not fetch reissue invoices', {
+      region: regionId,
+      error: {
+        name: error.name,
+        message: error.message,
+        stack: error.stack,
+        code: error.code
+      }
+    })
```

Comment:
Having just done [Better handle errors in Notifiers](https://github.com/DEFRA/water-abstraction-system/pull/273) the way the object is being created and passed to the notifier means we'd loose pino and Errbit seeing the original error.

I'm guessing you want to include the region ID in the packet to help when debugging. If this is the case we should be calling it like this now

```suggestion
    global.GlobalNotifier.omfg(
      'Could not fetch reissue invoices',
      { region: regionId },
      error
    )
```

Have a look at https://github.com/DEFRA/water-abstraction-system/blob/main/app/lib/base-notifier.lib.js#L83 and it should make more sense.